x-pack/plugins/aiops/public/application/utils/query_utils.ts

```diff
@@ -62,6 +77,12 @@ export function buildBaseFilterCriteria(
         ],
       },
     });
+  } else if (selectedGroup && !includeSelectedChangePoint) {
+    filterCriteria.push({
+      bool: {
+        must_not: [...groupFilter],
+      },
+    });
```

Comment:
`must_not` is OR by default, to get `groupFilter` treated with AND it needs to be nested in a bool filter again:

```
filterCriteria.push({
      bool: {
        must_not: [
          {
            bool: {
              filter: [...groupFilter],
            },
          },
        ],
      },
    });
```

Reply:
Updated in 655ffec956f353b5c6306d305719c1fa8788d48b